[ICL]Pr vdenc hevc

fulinjie opened the pull request.

fulinjie committed:
lavc/qsvenc_hevc: add VDENC support for HEVC

Signed-off-by: Linjie Fu <linjie.fu@intel.com>

lizhong1008 commented on libavcodec/qsvenc_hevc.c:
```diff
@@ -237,6 +237,9 @@ static const AVOption options[] = {
     { "main10",  NULL, 0, AV_OPT_TYPE_CONST, { .i64 = MFX_PROFILE_HEVC_MAIN10  }, INT_MIN, INT_MAX,     VE, "profile" },
     { "mainsp",  NULL, 0, AV_OPT_TYPE_CONST, { .i64 = MFX_PROFILE_HEVC_MAINSP  }, INT_MIN, INT_MAX,     VE, "profile" },
+#if QSV_HAVE_VDENC
+    { "low_power", "enable low power mode(experimental: many limitations by mfx version, BRC modes, etc.)", OFFSET(qsv.low_power), AV_OPT_TYPE_BOOL, { .i64 =  0 }, 0, 1, VE},
```
@fulinjie ,please  check Carl's comment, no to hide this option.

fulinjie replied:
I intended to match the behavior in VDENC H264.
If modified, I think same modification should be applied in H264 too.

lizhong1008 replied:
yes, please update H264 too to keep consistent.

fulinjie committed:
lavc/qsvenc: expose low_power option in H264 QSV

Always exposes low_power option for h264 qsv, and reports a warning
if VDENC is not supported in current version of MSDK.

Signed-off-by: Linjie Fu <linjie.fu@intel.com>

linjiefu commented:
https://patchwork.ffmpeg.org/patch/12448/

fulinjie commented:
Low power mode for hevc qsv is merged, close.
